src/nvim/ex_cmds.c

```diff
@@ -1583,22 +1583,43 @@ char *make_filter_cmd(char *cmd, char *itmp, char *otmp)
   if (otmp != NULL) {
     len += STRLEN(otmp) + STRLEN(p_srr) + 2;  // two extra spaces ("  "),
   }
+
+  const char *const cmd_args = strchr(cmd, ' ');
+  len += (is_pwsh && cmd_args)
+      ? STRLEN(" -ArgumentList ") + 2  // two extra quotes
+      : 0;
+
   char *const buf = xmalloc(len);
 
-#if defined(UNIX)
-  // Put delimiters around the command (for concatenated commands) when
-  // redirecting input and/or output.
   if (is_pwsh) {
     xstrlcpy(buf, "Start-Process ", len);
-    xstrlcat(buf, cmd, len);
+    if (cmd_args == NULL) {
+      xstrlcat(buf, cmd, len);
+    } else {
+      char *con = NULL;
+      char *dupcmd = xstrdup(cmd);
+      const char *const cmd_cmd = os_strtok(dupcmd, " ", &con);
+      xstrlcat(buf, cmd_cmd, len);
+      xstrlcat(buf, " -ArgumentList \"", len);
+      xstrlcat(buf, cmd_args, len);
+      xstrlcat(buf, "\"", len);
+      xfree(dupcmd);
```

Comment:
wait, why are we using strtok at all? We already got the pointer to `cmd_args` which tells us where the command name ends. So this becomes (not tested):

```suggestion
      xstrlcpy(buf + strlen(buf), cmd, cmd_arg - cmd);
      xstrlcat(buf, " -ArgumentList \"", len);
      xstrlcat(buf, cmd_args + 1, len);  // +1 to skip the leading space.
      xstrlcat(buf, "\"", len);
```